lib/core/src/recover/recoverer.rs

```diff
@@ -246,14 +247,27 @@ impl Recoverer {
                     }
                 }
                 Swap::Receive(receive_swap) => {
-                    let Some(recovered_data) = recovered_receive_data.get(swap_id) else {
+                    let Some(recovered_data) = recovered_receive_data.get_mut(swap_id) else {
                         log::warn!("Could not apply recovered data for Receive swap {swap_id}: recovery data not found");
                         continue;
                     };
-                    let timeout_block_height = receive_swap.timeout_block_height;
-                    let is_expired = liquid_tip >= timeout_block_height;
-                    if let Some(new_state) = recovered_data.derive_partial_state(is_expired) {
-                        receive_swap.state = new_state;
+                    if let Some(mrh_amount_sat) = recovered_data.mrh_amount_sat {
+                        receive_swap.payer_amount_sat = mrh_amount_sat;
+                        receive_swap.receiver_amount_sat = mrh_amount_sat;
+                    }
+                    let expected_lockup_amount_sat =
+                        receive_swap.receiver_amount_sat + receive_swap.claim_fees_sat;
+                    if receive_swap.lockup_tx_id.is_none() && recovered_data.lockup_tx_id.is_some()
+                    {
+                        let lockup_amount_sat =
+                            recovered_data.lockup_amount_sat.unwrap_or_default();
+                        if expected_lockup_amount_sat != lockup_amount_sat {
+                            error!(
+                                "Failed to verify lockup amount for Receive Swap {}: {} sat vs {} sat",
+                                swap_id, expected_lockup_amount_sat, lockup_amount_sat
+                            );
+                            recovered_data.lockup_tx_id = None;
```

Comment:
Do you think we can simplify the recoverer by not making it responsible for validating the lockup amount?
If we prevent claiming the wrong amount isn't that enough?

Reply:
The swap/payment would be set to Pending if we don't. Especially for non-local swaps

Reply:
Right but since this is an edge case (where the swOpers cheats) doesn't it worth the simplicity? Worst case the payment stays pending untill expiration.

Reply:
Simplified [0cb9ca1](https://github.com/breez/breez-sdk-liquid/pull/681/commits/0cb9ca189607c1fde820c6f104222accc8a879ce)